stellar_sdk/keypair.py

```diff
@@ -256,6 +258,54 @@ def from_mnemonic_phrase(
         )
         return cls.from_raw_ed25519_seed(raw_ed25519_seed)
 
+    @staticmethod
+    def generate_shamir_mnemonic_phrases(
+        member_threshold: int, member_count: int, passphrase: str = ""
+    ) -> list[str]:
+        """Generate mnemonic phrases using Shamir secret sharing method.
+
+        A randomly generated secret key is generated and split into `member_count`
+        mnemonic phrases. The secret key can be later reconstructed using any
+        subset of `member_threshold` phrases.
+
+        :param member_threshold: Number of members required to reconstruct the secret key.
+        :param member_count: Number of shares the secret is split into.
+        :param passphrase: An optional passphrase used to decrypt the secret key.
+        :return: A list of mnemonic phrases.
+        """
+        if member_count < member_threshold:
+            raise ValueError(
+                "There must be more members than the set threshold: "
+                "``member_count > member_threshold``."
+            )
+        secrets = Keypair.random().secret.encode()
+        return shamir_mnemonic.generate_mnemonics(
+            group_threshold=1,
```

Comment:
Only supporting a single group should be sufficient to meet the needs of the vast majority of users.